packages/payment-processor/src/payment/near-any-declarative.ts

```diff
@@ -0,0 +1,39 @@
+import { BigNumberish } from 'ethers';
+import { WalletConnection } from 'near-api-js';
+
+import { ClientTypes, ExtensionTypes } from '@requestnetwork/types';
+
+import { getRequestPaymentValues, validateRequest } from './utils';
+import { INearTransactionCallback, processNearPayment } from './utils-near';
+import { NearChains } from '@requestnetwork/currency';
+import { validatePaymentReference } from '../utils/validation';
+
+/**
+ * Processes the transaction to pay a declarative request with NEAR tokens.
+ * @param request the request to pay, must be declarative, with payment values on a Near network.
+ * @param walletConnection the Near provider.
+ * @param amount amount to pay, in NEAR tokens.
+ */
+export async function payNearAnyDeclarativeRequest(
+  request: ClientTypes.IRequestData,
+  walletConnection: WalletConnection,
+  amount: BigNumberish,
+  callback?: INearTransactionCallback,
+): Promise<void> {
+  validateRequest(request, ExtensionTypes.PAYMENT_NETWORK_ID.ANY_DECLARATIVE);
+
+  const { paymentReference, paymentAddress, network } = getRequestPaymentValues(request);
```

Comment:
I don't think this method will return this data.

th declarative-pn has a `paymentInfo` field in the `values` extension field.
And in case of off-chain conversion, it would contain:
```ts
  "paymentInfo": {
      "paymentAddress": "xxx.near",
      "paymentCurrency": "NEAR-near"
    },
```

The network property defined [here](https://github.com/RequestNetwork/requestNetwork/blob/ac361e8fd4e44f3c2638e315b84e7a80073f8e41/packages/payment-processor/src/payment/utils.ts#L92) won't return what you expect.
Similarly, you won't have a `paymentAddress`.

I think it would be best to export a method with low level parameter (`amount`, `address`, `paymentRef`, `version`) instead of the request object, otherwise we kind of add RF-specific logic within the protocol as the shape of the `paymentInfos` field is defined in RF but not in RN

Reply:
@leoslr , I did as you advised, thanks.